src/core/source_buffers/source_buffers_store.ts

```diff
@@ -239,11 +334,16 @@ export default class SourceBuffersStore {
    */
   public disposeAll() {
     POSSIBLE_BUFFER_TYPES.forEach((bufferType : IBufferType) => {
-      if (this.get(bufferType) != null) {
+      if (this.getStatus(bufferType).type === "set") {
         this.disposeSourceBuffer(bufferType);
       }
     });
   }
+
+  private _areNativeSourceBuffersReady() {
```

Comment:
If both are null, do we have to consider native source buffers as ready ?

Reply:
That's a good question.

If both are `null` at one end nothing will be able too play, but both would still be initialized.

What do you think we should do in that case, throw? never resolve?

Reply:
Maybe we could consider renaming the function into something more suggestive of "buffers are ready for playback". If both are null, then return false. Maybe it is clearer that way, no ?

Reply:
This also means that if we allow through our API to disable both, we will just wait forever at the `PeriodBuffer` step (so at `LOADING`). Is that what we want?

Reply:
The problem will exist in any case anyway, but we have to know what to do in that case.

Reply:
Done for the renaming + new behavior, what do you think ?